src/tasks/gitea/gitea-drone-oauth.test.ts

```diff
@@ -0,0 +1,146 @@
+import { expect } from 'chai'
+import {
+  DroneSecret,
+  getGiteaAuthorizationHeaderCookies,
+  GiteaDroneError,
+  isSecretValid,
+  getCookieValue,
+} from './gitea-drone-oauth'
+import { OAuth2Application } from '@redkubes/gitea-client-node'
+import sinon from 'sinon'
+import axios from 'axios'
+
+describe('Gitea-Drone: Validate Secrets', () => {
+  let consoleLog: sinon.SinonStub
+  let sandbox: sinon.SinonSandbox
+  beforeEach(() => {
+    sandbox = sinon.createSandbox()
+    consoleLog = sandbox.stub(console, 'log')
+  })
+  afterEach(() => {
+    sandbox.restore()
+  })
+  it('should successfully validate secret', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: Buffer.from('def').toString('base64'),
+    }
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abc',
+      },
+    ]
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.true
+  })
+  it('should fail to validate on no secret', () => {
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abc',
+      },
+    ]
+    expect(isSecretValid(undefined, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith('Remote secret was not found')).to.be.true
+  })
+  it('should fail validation when the secret contains invalid data', () => {
```

Comment:
again, please remove all tests related to the function that has should not exist in the first place: `isSecretValid`